mypy/checkexpr.py

```diff
@@ -2822,9 +2822,17 @@ def check_boolean_op(self, e: OpExpr, context: Context) -> Type:
         # inference of the right operand so that expressions such as
         # '[1] or []' are inferred correctly.
         ctx = self.type_context[-1]
-        left_type = self.accept(e.left, ctx)
+        left_type = get_proper_type(self.accept(e.left, ctx))
+        keep_none = (
+            not self.chk.options.strict_optional
+            and isinstance(left_type, UnionType)
+            and any(
+                isinstance(get_proper_type(t), NoneType)
+                for t in left_type.items
+            )
+        )
         expanded_left_type = try_expanding_sum_type_to_union(
-            self.accept(e.left, ctx), "builtins.bool"
+            left_type, "builtins.bool", keep_none=keep_none,
         )
```

Comment:
```suggestion
            left_type, "builtins.bool", keep_none=True,
        )
```

> Yes, we do. Because we only need this logic to be executed at some use-cases. If I remove this param, we will have 5-6 failing tests.

Would this suffice then?

Reply:
Maybe I was too strict with this. I will try just `True`, good point! 👍

Reply:
Done! Thanks!

Reply:
No problem! :)